### PR TITLE
SNOW-1524269 native put get support

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-hx**
teamwork issue 1035

### CODE REVIEW

File: cpp/StatementPutGet.cpp [hunk 1]
```diff
@@ -3,11 +3,35 @@
...
 #include "StatementPutGet.hpp"
+#include "curl_desc_pool.h"
 
 using namespace Snowflake::Client;
 
+static size_t file_get_write_callback(char* ptr, size_t size, size_t nmemb, void* userdata)
+{
+  size_t data_size = size * nmemb;
+  std::basic_iostream<char>* recvStream = (std::basic_iostream<char>*)(userdata);
+  if (recvStream)
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-hx) [reply cpp/StatementPutGet.cpp:17]: This is the behavior copied from ODBC. https://github.com/snowflakedb/snowflake-odbc/blob/master/Source/Platform/RestResponse.hpp#L214
for GET, there could be header only request to retrieve the metadata of the source file, in that case `recvStream` would be null.

File: cpp/lib/ResultSetPutGet.cpp [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+{
+namespace Client
+{
+ResultSetPutGet::ResultSetPutGet(ITransferResult *result) :
+    ResultSet(SF_PUTGET_FORMAT),
+    m_cmdType(result->getCommandType())
+{
+  while (result->next())
+  {
+    std::vector<std::string> row;
```
[the remark this answers is not in the document]
Comment (sfc-gh-ext-simba-hx) [reply cpp/lib/ResultSetPutGet.cpp:83]: done

**Review by sfc-gh-dstempniak — APPROVED** [review]
LGTM

**Comment by sfc-gh-dprzybysz** [comment]
This PR contains changes from https://github.com/snowflakedb/libsnowflakeclient/pull/724
let's merge #724 before this one to clean up changes

**Pull request merged**